docs/projectTypes/updatetonavigationview.md

```diff
@@ -0,0 +1,225 @@
+
+### C# code you will have to modify:
+ - Add the navigationView control to ViewModel initialization.
+
+The resulting code should look like this:
+ ```csharp
+public sealed partial class ShellPage : Page
+{
+    public ShellViewModel ViewModel { get; } = new ShellViewModel();
+
+    public ShellPage()
+    {
+        InitializeComponent();
+        DataContext = ViewModel;
+        ViewModel.Initialize(shellFrame, navigationView);
+    }
+}
+```
+
+## 4. Add NavigationViewItemExtensions.cs
+
+Add this extension class in the **Helpers** folder to the project. This allows the NavigationViewItems to contain a Type property that is used for navigation.
+```csharp
+public class NavigationViewItemExtensions
+{
+    public static Type GetPageType(NavigationViewItem obj)
+    {
+        return (Type)obj.GetValue(PageTypeProperty);
+    }
+
+    public static void SetPageType(NavigationViewItem obj, Type value)
+    {
+        obj.SetValue(PageTypeProperty, value);
+    }
+
+    public static readonly DependencyProperty PageTypeProperty =
+        DependencyProperty.RegisterAttached("PageType", typeof(Type),
+        typeof(NavigationViewItemExtensions), new PropertyMetadata(null));
+}
+```
+
+## 5. Update ShellViewModel.cs
+ShellViewModel's complexity will be reduced significantly, these are the changes that you will have to implement on the class.
+### C# code you will have to remove:
+ - private **const properties** for Visual States (Panoramic, Wide, Narrow).
+ - **IsPaneOpen** observable property.
+ - **DisplayMode** observable property.
+ - **ObservableCollections** properties for **PrimaryItems** and **SecondaryItems**.
+ - **OpenPaneCommand** and handler method.
+ - **ItemSelectedCommand** and handler method.
+ - **StateChangedCommand** and handler method.
+ - **GoToState** method.
+ - **PopulateNavItems** method and method call from Initialize.
+
+### C# code you will have to add _(implementation below)_:
+ - **ItemInvokedCommand** and handler method.
+  - **IsNavigationViewItemFromPageType** method.
+
+### C# code you will have to update _(implementation below)_:
+ - **Initialize** method.
+ - **Frame_Navigated** method with the implementation below.
+
+ The resulting code should look like this:
+```csharp
+public class ShellViewModel : Observable
+{
+    private NavigationView _navigationView;
+    private object _selected;
+    private ICommand _itemInvokedCommand;
+
+    public object Selected
+    {
+        get { return _selected; }
+        set { Set(ref _selected, value); }
+    }
+
+    public ICommand ItemInvokedCommand => _itemInvokedCommand ?? (_itemInvokedCommand = new RelayCommand<NavigationViewItemInvokedEventArgs>(OnItemInvoked));
+
+    public ShellViewModel()
+    {
+    }
+
+    public void Initialize(Frame frame, NavigationView navigationView)
+    {
+        _navigationView = navigationView;
+        NavigationService.Frame = frame;
```

Comment:
Where does the NavigationService property originate from? Previously this was injected using dependency injection with the Prism framework